core/src/lib.rs

```diff
@@ -291,12 +297,60 @@ impl Schema {
     pub fn identifies_attribute(&self, x: &NamespacedKeyword) -> bool {
         self.get_entid(x).map(|e| self.is_attribute(e)).unwrap_or(false)
     }
+
+    /// Returns an edn::Value::Vector representation of the schema map.
+    pub fn to_edn_value(&self) -> edn::Value {
+        let mut all_values = Vec::new();
+        for (entid, attribute) in &self.schema_map {
+            let mut attribute_map: BTreeMap<edn::Value, edn::Value> = BTreeMap::default();
+            let some_ident = self.get_ident(entid.clone());
+            if some_ident.is_some() {
+                let ident = some_ident.unwrap();
+                attribute_map.insert(values::DB_IDENT.clone(), edn::Value::NamespacedKeyword(ident.clone()));
+            } else {
+                attribute_map.insert(values::DB_IDENT.clone(), edn::Value::Keyword(Keyword::new(entid.to_string())));
+            }
+
+            let value_type = format!("{:?}", attribute.value_type);
+            attribute_map.insert(values::DB_VALUE_TYPE.clone(), edn::Value::NamespacedKeyword(NamespacedKeyword::new("db.type", &value_type.to_lowercase())));
+
+            attribute_map.insert(values::DB_CARDINALITY.clone(), if attribute.multival { values::DB_CARDINALITY_MANY.clone() } else { values::DB_CARDINALITY_ONE.clone() });
+
+            if attribute.unique == Some(attribute::Unique::Value) {
+                attribute_map.insert(values::DB_UNIQUE.clone(), values::DB_UNIQUE_IDENTITY.clone());
+            }
+            
+            if attribute.index {
+                attribute_map.insert(values::DB_INDEX.clone(), edn::Value::Boolean(true));
+            }
+
+            if attribute.fulltext {
+                attribute_map.insert(values::DB_FULLTEXT.clone(), edn::Value::Boolean(true));
+            }
+
+            if attribute.component {
+                attribute_map.insert(values::DB_IS_COMPONENT.clone(), edn::Value::Boolean(true));
+            }
+
+            all_values.push(edn::Value::Map(attribute_map));
```

Comment:
Rather than doing this with mutable vecs, use comprehensions:

```rust
pub fn to_edn_value(&self) -> edn::Value {
    edn::Value::Vector(
        self.schema_map
            .map(|(entid, attribute)| self.schema_to_edn(entid, attribute))
            .collect())
}
```